### PR TITLE
Fix regression on windows: Env var names are case insensitive

### PR DESCRIPTION
**Pull request by autoantwort**
Fixes https://github.com/microsoft/vcpkg/issues/41199

Thank you @tmp64 for identifying why the problem exists :)

[The old code](https://github.com/microsoft/vcpkg-tool/commit/d3fc35774f86782f356ef0f9352ba857fa57302e#diff-33cf7c311a76d4a838f91b078c2f8cbc984557379f7b345a268ec6deb665a91eL672) was also case insensitive, because `get_environment_variable` is case insensitive

### CODE REVIEW

File: src/vcpkg/base/system.process.cpp [hunk 1]
```diff
@@ -661,8 +661,9 @@ namespace vcpkg
         {
             auto pos = env_var.find('=');
             auto key = env_var.substr(0, pos);
-            if (Util::Sets::contains(env_strings, key) ||
-                Util::any_of(env_prefix_string, [&](auto&& group) { return Strings::starts_with(key, group); }))
+            if (Util::Sets::contains(env_strings, key) || Util::any_of(env_prefix_string, [&](auto&& group) {
```
Comment (tmp64) [inline src/vcpkg/base/system.process.cpp:664]: Is `Util::Sets::contains` case-insensitive? `SystemRoot` is in `env_strings`

Comment (autoantwort) [reply src/vcpkg/base/system.process.cpp:664]: Good catch. It too late for me 😅 Will fix that :)

**Comment by BillyONeal** [comment]
Since this blew up in an important way, can you add a test?

**Comment by Neumann-A** [comment]
Seems to fix https://github.com/microsoft/vcpkg/issues/41235

**Comment by BillyONeal** [comment]
Thanks for the superb diagnostics @tmp64 and the fix @autoantwort

**Pull request merged**